### PR TITLE
[Recipes] Matching the metric names of recipes and mlflow.evaluate

### PR DESCRIPTION
**Pull request by sunishsheth2009**
## What changes are proposed in this pull request?
[Recipes] Matching the metric names of recipes and mlflow.evaluate

## How is this patch tested?
- [x] Test passes

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [x] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/recipes/utils/metrics.py [hunk 1]
```diff
@@ -42,20 +42,20 @@ def from_custom_metric_dict(cls, custom_metric_dict):
...
 
 BUILTIN_MULTICLASS_CLASSIFICATION_RECIPE_METRICS = [
-    RecipeMetric(name="recall_score", greater_is_better=True),
-    RecipeMetric(name="precision_score", greater_is_better=True),
+    RecipeMetric(name="recall", greater_is_better=True),
+    RecipeMetric(name="precision", greater_is_better=True),
     RecipeMetric(name="f1_score_macro", greater_is_better=True),
     RecipeMetric(name="f1_score_micro", greater_is_better=True),
-    RecipeMetric(name="accuracy_score", greater_is_better=True),
-    RecipeMetric(name="roc_auc_ovr", greater_is_better=True),
```
Comment (jinzhang21) [inline mlflow/recipes/utils/metrics.py:58]: Could you file a ticket to add `roc_auc_score` to both mlflow.evaluate() and recipes? This is a useful metric for classification. We should support it: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html

File: tests/recipes/test_train_step.py [hunk 2]
```diff
@@ -300,7 +300,7 @@ def test_train_step_classifier_automl(
                     - lgbm
         """.format(
             tracking_uri=mlflow.get_tracking_uri(),
-            metric="roc_auc" if recipe == "classification/binary" else "roc_auc_ovr",
+            metric="f1_score" if recipe == "classification/binary" else "f1_score_micro",
```
Comment (jinzhang21) [inline tests/recipes/test_train_step.py:303]: `f1_score_micro` isn't particularly useful, -> `f1_score_macro` instead.

**Review by jinzhang21 — APPROVED** [review]
Approved provided the comments are addressed.

File: mlflow/recipes/utils/metrics.py (hunk 1 above)
Comment (sunishsheth2009) [reply mlflow/recipes/utils/metrics.py:58]: Seems like we already support it. So added it back :) 
It is just named differently. 
https://github.com/mlflow/mlflow/blob/a2783d058d2f3151421565af2146bd4f455acfcf/mlflow/models/evaluation/default_evaluator.py#L237

File: mlflow/recipes/steps/automl/flaml.py [hunk 3]
```diff
@@ -138,6 +140,10 @@ def _create_model_automl(
     try:
         if primary_metric in _MLFLOW_TO_FLAML_METRICS and primary_metric in evaluation_metrics:
             metric = _MLFLOW_TO_FLAML_METRICS[primary_metric]
+            if metric in _MULTICLASS_TO_FLAML_METRICS and extended_task in [
+                "classification/multiclass"
+            ]:
+                metric = _MULTICLASS_TO_FLAML_METRICS[metric]
```
Comment (jinzhang21) [inline mlflow/recipes/steps/automl/flaml.py:146]: I don' think it's necessary to introduce `_MULTICLASS_TO_FLAML_METRICS` since there's only 1 item required for special handling. `roc_auc` is already defined in `_MLFLOW_TO_FLAML_METRICS`. You can probably do something like
```
if primary_metric == `roc_auc` and extended_task == "classification/multiclass":
  metric = roc_auc_ovr
```

File: mlflow/recipes/utils/metrics.py (hunk 1 above)
Comment (jinzhang21) [reply mlflow/recipes/utils/metrics.py:58]: This is great!

**Review by jinzhang21 — APPROVED** [review]
